rfcs/text/0000-alternative-planning.md

```diff
@@ -0,0 +1,315 @@
+described in the Roadmaps and Planning RFC and as an evolution of the current process, will not fit DCP organizational and development realities.
+
+This RFC tries to concisely describe those problems and propose alternative solutions. It comes as a result of being
+asked to write these alternatives up. Due to time constraints it's 
+nowhere near as comprehensive, well-written or referenced as I would like. Nonetheless, I hope it can help spur 
+constructive discussion.
+
+This RFC is the author's own point of view as a technical lead and someone with long experience in highly distributed 
+open-source projects that require co-operation between peer organizations and individuals.
+
+## Detailed Design
+
+From my understanding, the roadmaps and planning RFC describes a 70 day process where scientists, project leads, product owners, user experience
+experts and technical leads create a roadmap consisting of themes, each theme having one of more objectives. These 
+objectives are further described by user stories. Each objective is represented in the DCP Zenhub board as an epic with
+user stories represented by an item attached to the epic. Further items can be created as spikes to spend time clarifying
+the user stories and/or performing design work prior to implementation. Each item is assigned a milestone where there are
+3 milestones of equal length in a quarter.
+
+In my experience of the existing planning process almost all spikes are cross-component issues. DCP culture pretty much
+ mandates that these require an RFC. An RFC
+takes a minimum of 3 weeks from inception to approval, though realistically RFCs of any complexity require considerably
+more time to understand the problem and align with other planned and existing technical work.
+
+From my experience so far, if an RFC is required this means that it will take a minimum of 2 milestones (8 weeks)
+before it can be considered complete. A T-shirt estimate is only possible once the RFC is approved and there is a
+strong chance that estimate will be more than the remainder of the quarter (4 weeks). Which means it takes an entire
+quarter to plan work and you can only start it in the next quarter at the earliest. Even then, the roadmap process for
+that next quarter will have started long before so I fear that in the very worst case it could be *up to 9 months between planning and implementation*.
+In other words, we will have a waterfall of roadmapping, planning and implementation with an extremely long cycle time.
+
+![](../images/0000-process-cartoon.png)
+
+I have grave doubts about doing down this road for two reasons:
+
+* **Prediction is hard** - predicting what software you need to create is hard. It's harder the further out you try. 
+Surprisingly soon even the roughest prediction is impossible.
+* **Science is unpredictable** - the requirements for the HCA shift. The science changes unpredictably.
```

Comment:
Can you suggest concrete examples? Is this mitigated by the Science Roadmap which informs the Product Roadmap?

Reply:
I haven't seen the science roadmap. You could provide a link but frankly I may not have the knowledge to make a judgement. All I know is that I see constant requirements churn and reprioritization at my level coming from scientific changes. These aren't only the science but also the organization of it. The CZI initiative, for example, which came along at short notice and is displacing Q3 work.

Reply:
Jonah presented the science roadmap at the last DCP F2F. By _CZI initiative_, do you mean the _DCP Data Submission Incentive_?

Reply:
Naturally I meant the DCP incentive. And I've had my own exchange with @jonahcool where I think the north stars are great but we have instituted no way of measuring whether we're meeting them and we have roadmap items that have not been linked to any north star, measurable or otherwise.

Reply:
The hope is the longterm road map will help smooth some of this out. A scientific roadmap is also a part of the solution, and honestly better stakeholder management on the part of the PLs. We should strive to be reactive to changes in stakeholder needs (at the consortium and user level) but having clear guidance important. This is great feedback.